rawfile/filesystem/btrfs.py

```diff
@@ -0,0 +1,180 @@
+        options: list[str] = [],
+    ) -> str:
+        if mountpoint is None:
+            if self.mountpoint is None:
+                raise InvalidMountpointError()
+            mountpoint = self.mountpoint
+        options.extend(["-o", "flushoncommit"])
+        self.mountpoint = mountpoint
+        try:
+            current_dev = get_device_for_mountpoint(mountpoint)
+            if current_dev == self.device:
+                return f"Device {self.device} already mounted at {mountpoint}"
+            elif current_dev:
+                self.unmount()
+            Path(mountpoint).mkdir(exist_ok=True)
+            output = subprocess.run(
+                f"mount -t {self.__filesystem__} {' '.join(options)} {self.device} {mountpoint}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemMountError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemMountError(self.__filesystem__, str(e))
+
+    def unmount(self, clear_mountpoint=False) -> str:
+        try:
+            if get_device_for_mountpoint(self.mountpoint) != self.device:
+                return f"{self.device} is not mounted at {self.mountpoint}"
+            output = subprocess.run(
+                f"umount {self.mountpoint}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            if clear_mountpoint:
+                # NOTE: using rmdir since unmounted path should be empty
+                Path(self.mountpoint).rmdir()
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemUnmountError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemUnmountError(self.__filesystem__, str(e))
+
+    def resize(self) -> str:
+        try:
+            output = subprocess.run(
+                f"btrfs filesystem resize max {self.mountpoint}",
+                shell=True,
+                check=True,
+                capture_output=True,
+            )
+            return output.stdout.decode()
+        except subprocess.CalledProcessError as e:
+            raise FileSystemResizeError(self.__filesystem__, e.stderr.decode())
+        except Exception as e:
+            raise FileSystemResizeError(self.__filesystem__, str(e))
+
+    def create_snapshot(self, name: str):
+        try:
+            with TemporaryDirectory(
+                prefix=f"rawfile.{self.__filesystem__}.fs-snapshot."
+            ) as tmp_dir:
+                try:
+                    snapshots_dir = f"{tmp_dir}/.snapshots"
+                    snapshot_subvol = f"{snapshots_dir}/{name}"
+                    subprocess.run(
+                        f"""
+                        set -ex
```

Comment:
Probably safer to set pipefail?

```suggestion
                        set -exo pipefail
```